src/util.h

```diff
@@ -258,14 +262,41 @@ inline void MakeTmpdir(std::string* path) {
   *path = std::string(tmppath_buffer) + "gflags_unittest";
   _mkdir(path->c_str());
 }
-#else
+#elif defined(HAVE_SYS_STAT_H)
 inline void MakeTmpdir(std::string* path) {
   if (!path->empty()) {
 	int err = mkdir(path->c_str(), 0755);
 	if (err == 0 || errno == EEXIST) return;
   }
   mkdir("/tmp/gflags_unittest", 0755);
 }
+#elif __cplusplus >= 201703L
+inline void MakeTmpdir(std::string* path) {
+  if (!path->empty()) {
+    char *c = mkdtemp(path->data());
+    if (c != nullptr || errno == EEXIST) return;
+  }
+  mkdtemp(path->data());
```

Comment:
This behavior differs from the other alternatives. This line is reached when `path` is an empty string!